src/vic3_world/laws/laws_importer_tests.cpp

```diff
@@ -0,0 +1,156 @@
+   input << "\tactive = yes\n";
+   input << "}\n";
+   input << "1 ={ \n";
+   input << "\tlaw = test_law_two	country = 1\n";
+   input << "\tactive = yes\n";
+   input << "}\n";
+   input << "2 = {\n";
+   input << "\tlaw = test_law_one	country = 2\n";
+   input << "\tactive = yes\n";
+   input << "}\n";
+   input << "3 = {\n";
+   input << "\tlaw = test_law_three	country = 2\n";
+   input << "\tactive = yes\n";
+   input << "}\n";
+   input << "\t}\n";
+   input << "}\n";
+
+   const std::map<int, std::set<std::string>> laws = ImportLaws(input);
+
+   EXPECT_THAT(laws,
+       testing::UnorderedElementsAre(testing::Pair(1, std::set<std::string>{"test_law_one", "test_law_two"}),
+           testing::Pair(2, std::set<std::string>{"test_law_one", "test_law_three"})));
+}
+
+
+TEST(Vic3WorldLawsLawsImporter, LawsSetAsNoneAreSkipped)
+{
+   std::stringstream input;
+   input << "={\n";
+   input << "\tdatabase={\n";
+   input << "0 = none\n";
+   input << "\t}\n";
+   input << "}\n";
+
+   const std::map<int, std::set<std::string>> laws = ImportLaws(input);
+
+   EXPECT_TRUE(laws.empty());
+}
+
+
+TEST(Vic3WorldLawsLawsImporter, InactiveLawsAreSkipped)
+{
+   std::stringstream input;
+   input << "={\n";
+   input << "\tdatabase={\n";
+   input << "0 = {\n";
+   input << "\tlaw = test_law_one	country = 1\n";
+   input << "}\n";
+   input << "\t}\n";
+   input << "}\n";
+
+   const std::map<int, std::set<std::string>> laws = ImportLaws(input);
+
+   EXPECT_TRUE(laws.empty());
+}
+
+
+TEST(Vic3WorldLawsLawsImporter, LawsWithOddActiveBlockAreLogged)
+{
+   std::stringstream log;
+   std::streambuf* cout_buffer = std::cout.rdbuf();
+   std::cout.rdbuf(log.rdbuf());
+
+   std::stringstream input;
+   input << "={\n";
+   input << "\tdatabase={\n";
+   input << "0 = {\n";
+   input << "\tlaw = test_law_one	country = 1\n";
+   input << "\tactive=weirdness\n";
+   input << "}\n";
+   input << "\t}\n";
+   input << "}\n";
+
+   const std::map<int, std::set<std::string>> _ = ImportLaws(input);
+
+   std::cout.rdbuf(cout_buffer);
+
+   EXPECT_THAT(log.str(),
+       testing::HasSubstr(
+           R"([WARNING] Law had odd active block. Please report to the converters team and upload your save.)"));
+}
+
+
+TEST(Vic3WorldLawsLawsImporter, LawsWithoutCountryAreLogged)
+{
+   std::stringstream log;
+   std::streambuf* cout_buffer = std::cout.rdbuf();
+   std::cout.rdbuf(log.rdbuf());
+
+   std::stringstream input;
+   input << "={\n";
+   input << "\tdatabase={\n";
+   input << "0 = {\n";
+   input << "\tlaw = test_law_one\n";
+   input << "\tactive = yes\n";
+   input << "}\n";
+   input << "\t}\n";
+   input << "}\n";
+
+   const std::map<int, std::set<std::string>> _ = ImportLaws(input);
+
+   std::cout.rdbuf(cout_buffer);
+
+   EXPECT_THAT(log.str(),
+       testing::HasSubstr(
+           R"([WARNING] Law without a country. Please report to the converters team and upload your save.)"));
+}
+
+
+TEST(Vic3WorldLawsLawsImporter, LawsWithoutNamesAreImported)
```

Comment:
I'm unsure what's the use-case for this or when would it happen. Maybe if someone disabled some mod midrun....

Reply:
It really shouldn't, but I decided to be explicit what happens in that case.